wordpress-mobile/WordPress-iOS
# [Tooling] Cleanup files scanned for NSLocalizedString

### What

 - Updates the list of excluded files during the `generate_strings_file_for_glotpress` lane so that it excludes Unit Tests and uses more generic glob patterns
 - Fixes the `I18n.swift` file which was referencing `NSLocalizedString` while it shouldn't use it when it's a use case like here not intended to be extracted by `genstrings`.
 - Fixes a unit test using `NSLocalizedString` on a string format that was not localized in the tested app code anyway.
 - Updates the `release-toolkit` to `2.3.0` now that it's out (because my `bundle install` was complaining about `develop` when I tried to test those changes otherwise)

### To Test

 - Create a test branch from this one
 - Run `bundle exec fastlane generate_strings_file_for_glotpress`
 - In the commit that was created, observe that the `WordPress/Resources/en.lproj/Localizable.strings` file has a couple of keys or comments removed
 - For each removed comment or key, search all the occurrences of the corresponding key in the source code and confirm that the instances that were removed were only present in the Unit Tests.
    - For example, for the `"%@ comments"` key, the `Number of Comments` translator hint got removed from the `.strings` file. but we can confirm that [the only instance with that comment for that key is in the `ReaderPostCardCellTests.swift`](https://github.com/wordpress-mobile/WordPress-iOS/blob/develop/WordPress/WordPressTest/ReaderPostCardCellTests.swift#L164) and was thus only used for unit tests.
    - Meanwhile, the only 2 other instances are both used in the app code — [here](https://github.com/wordpress-mobile/WordPress-iOS/blob/develop/WordPress/Classes/ViewRelated/Blog/Site%20Settings/DiscussionSettingsViewController.swift#L529) and [here](https://github.com/wordpress-mobile/WordPress-iOS/blob/develop/WordPress/Classes/ViewRelated/Reader/ReaderPostCardCell.swift#L850) — and the comments for those two are still present on that key in the generated `.strings` file.
   - Likewise, for the `"%@, %@"` key which got completely removed (comment, key and translation) from the `.strings` file, we can confirm that [the only call sites using that key were all in `PostTests.swift`](https://github.com/wordpress-mobile/WordPress-iOS/blob/develop/WordPress/WordPressTest/PostTests.swift#L324-L343) and nowhere used in the app itself, which is why it's safe to remove them from translation. In fact, I've updated that test in this PR because in the corresponding app code this test is testing, [the `%@, %@` format string itself isn't translated (but its individual components are)](https://github.com/wordpress-mobile/WordPress-iOS/blob/develop/WordPress/Classes/Models/Post.swift#L281-L287).
 - Delete your test branch
